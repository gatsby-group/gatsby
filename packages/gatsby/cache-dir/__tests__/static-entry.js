import React from 'react'
import DevelopStaticEntry from '../develop-static-entry'

jest.mock(`fs`)

jest.mock(
  `../sync-requires`,
  () => {
    return {
      components: {
        'page-component---src-pages-test-js': () => null,
      },
    }
  },
  { virtual: true }
)

jest.mock(
  `../data.json`,
  () => {
    return {
      dataPaths: [{
        [`about.json`]: `/400/about`,
      }],
      pages: [{
          path: `/about/`,
          componentChunkName: `page-component---src-pages-test-js`,
          jsonName: `about.json`,
      }],
    }
  },
  { virtual: true }
)


const MOCK_FILE_INFO = {
  [`${process.cwd()}/public/webpack.stats.json`]: `{}`,
}

require(`fs`).__setMockFiles(MOCK_FILE_INFO)

// Needs to be imported after __setMockFiles is called, and imports get hoisted.
const StaticEntry = require(`../static-entry`).default

const reverseHeadersPlugin = {
  plugin: {
    onPreRenderHTML: ({ getHeadComponents, replaceHeadComponents }) => {
      const headComponents = getHeadComponents()
      headComponents.reverse()
      replaceHeadComponents(headComponents)
    },
  },
}

const fakeStylesPlugin = {
  plugin: {
    onRenderBody: ({ setHeadComponents }) => setHeadComponents([
      <style key="style1">.style1 {}</style>,
      <style key="style2">.style2 {}</style>,
      <style key="style3">.style3 {}</style>,
    ]),
  },
}

describe(`develop-static-entry`, () => {
  test(`onPreRenderHTML can be used to replace head components`, (done) => {
    global.plugins = [
      fakeStylesPlugin,
      reverseHeadersPlugin,
    ]

    DevelopStaticEntry(`/about/`, (_, html) => {
      expect(html).toMatchSnapshot()
      done()
    })
  })
})

describe(`static-entry`, () => {
  beforeEach(() => {
    global.__PATH_PREFIX__ = ``
  })

  test(`onPreRenderHTML can be used to replace head components`, (done) => {
    global.plugins = [
      fakeStylesPlugin,
      reverseHeadersPlugin,
    ]

    StaticEntry(`/about/`, (_, html) => {
      expect(html).toMatchSnapshot()
      done()
    })
  })
})
