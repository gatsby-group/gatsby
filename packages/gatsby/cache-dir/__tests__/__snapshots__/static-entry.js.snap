// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`develop-static-entry onPreRenderHTML can be used to replace header components 1`] = `"<!DOCTYPE html><html><head><meta charSet=\\"utf-8\\"/><meta http-equiv=\\"x-ua-compatible\\" content=\\"ie=edge\\"/><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, shrink-to-fit=no\\"/><style>.style2 </style><style>.style1 </style><script src=\\"/socket.io/socket.io.js\\"></script></head><body><div id=\\"___gatsby\\"></div><script src=\\"/commons.js\\"></script></body></html>"`;

exports[`static-entry onPreRenderHTML can be used to replace header components 1`] = `"<!DOCTYPE html><html><head><meta charSet=\\"utf-8\\"/><meta http-equiv=\\"x-ua-compatible\\" content=\\"ie=edge\\"/><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, shrink-to-fit=no\\"/><style>.style2 </style><style>.style1 </style></head><body><div id=\\"___gatsby\\"></div><script id=\\"gatsby-script-loader\\">/*<![CDATA[*/window.page={\\"path\\":\\"/about/\\",\\"componentChunkName\\":\\"page-component---src-pages-test-js\\",\\"jsonName\\":\\"about.json\\"};/*]]>*/</script></body></html>"`;
